### PR TITLE
Separates data arrays into contenttype- and fielddefinitionobjects

### PR DESCRIPTION
**Pull request by htollefsen**
Adds a more strict, but accurate way to build up creations. Only identifiers are required. No need for the array-buildup to be so loose, when the accepted/possible data is so strict.

### CODE REVIEW

File: src/Transfer/EzPlatform/Data/FieldDefinitionObject.php [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+     */
+    public function getNames()
+    {
+        return $this->names;
+    }
+
+    /**
+     * @param FieldDefinitionCreateStruct $fieldDefinitionCreateStruct
+     */
+    public function dsa(&$fieldDefinitionCreateStruct)
```
Comment (htollefsen) [inline src/Transfer/EzPlatform/Data/FieldDefinitionObject.php]: Ummm..

File: composer.json [hunk 2]
```diff
@@ -16,11 +17,11 @@
...
     "require": {
         "php": ">=5.4.4",
-        "transfer/transfer": "~1.0",
-        "ezsystems/ezpublish-kernel": "dev-master"
+        "transfer/transfer": "~1.0.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.7"
+        "phpunit/phpunit": "^4.7",
+        "ezsystems/ezpublish-kernel": "dev-master"
```
Comment (ghost) [inline composer.json]: @htollefsen Should the `ezpublish-kernel` be in the dev section?

Comment (htollefsen) [inline composer.json]: You're right. Fixed in updated PR.

**Comment by ghost** [comment]
Thanks, @htollefsen!

**Pull request merged**